### ISSUE TITLE
📦 Packages with vulnerability

### ISSUE THREAD

**Issue by t1gu1** [tinacms/tinacms#4474]
### Describe the issue that you're seeing. Any Loom videos or screenshots usually help a lot!

Some package contain vulnerabilities.

![image](https://github.com/tinacms/tinacms/assets/12479055/e78ab37a-493b-41de-bc04-3ef7d4f92fc5)

### Reproduction

localhost

### Steps to reproduce

- npm install
- npm audit

### System Info

```shell
- Mac OS
- Node ➜ v18.16.0
- Node ➜ 9.5.1
- tinacms ➜ 1.6.0
- tinacms/cli ➜ 1.5.42
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/tinacms/tinacms/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://tina.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/tinacms/tinacms/issues) that reports the same bug.

**Comment by exofoliohq** [comment]
I see the same on my end with :
```
- Windows 11
- Node ➜ v20.12.0
- tinacms ➜ 1.6.1
- tinacms/cli ➜ 1.5.43
```
Terminal output : 
```
# npm audit report

axios  0.8.1 - 0.27.2
Severity: moderate
Axios Cross-Site Request Forgery Vulnerability - https://github.com/advisories/GHSA-wf5p-g6vw-rhxx
No fix available
node_modules/axios
  @tinacms/cli  *
  Depends on vulnerable versions of @tinacms/app
  Depends on vulnerable versions of axios
  Depends on vulnerable versions of tinacms
  node_modules/@tinacms/cli

lodash.set  *
Severity: high
Prototype Pollution in lodash - https://github.com/advisories/GHSA-p6mc-m468-83gw
No fix available
node_modules/lodash.set
  tinacms  <=0.0.0-20240328200248 || 0.4.0-dev.0 || >=0.50.0
  Depends on vulnerable versions of lodash.set
  node_modules/tinacms
    @tinacms/app  <=0.0.22 || >=1.2.0
    Depends on vulnerable versions of tinacms
    node_modules/@tinacms/app

5 vulnerabilities (1 moderate, 4 high)

Some issues need review, and may require choosing
a different dependency.
```
May bumping up these dependencies be simple, straightforward and break nothing 🤞

**Comment by ncn-ssw** [comment]
Parked in Sprint 3 due to onboarding issues

**Comment by Ben0189** [comment]
As per my investigation here are the number of vulnerabilities packages



![Image](https://github.com/user-attachments/assets/0bb3e288-2fa3-44ee-9659-87f3e60dbdf2)

**Comment by Ben0189** [comment]
loadash.set is unsed in tinacms package where no longer maintained as per this [article](https://security.snyk.io/package/npm/lodash.set/4.3.2)

**Comment by Ben0189** [comment]
Vulnerabilities down from 26 -> 6

![Image](https://github.com/user-attachments/assets/9dbcec6b-c328-4f80-ba75-c1794b320361)
**Figure: current vulnerable packages**

Left over vulnerable packages: 
Plate - which will be removed after 

![Image](https://github.com/user-attachments/assets/8f6804d8-dfdc-49b4-9d13-ee8e7625e7d6)
**Figure: plate which will be removed once plate upgrade PBI is done**


![Image](https://github.com/user-attachments/assets/fba2bd30-b9c6-4a31-a23e-7e84a6fe7815)
**Figure: Next js in important packages**

**Comment by exofoliohq** [comment]
Congrats 👏👏👏

**Comment by istr** [comment]
For the time being you might want to try this workaround in `package.json` (for `npm`):

```json
  "overrides": {
    "tinacms@2.2.0": {
      "lodash.set": "npm:lodash@4.17.21",
      "@tinacms/search": {
        "@tinacms/graphql": {
          "lodash.set": "npm:lodash@4.17.21",
          "micromatch": "4.0.8"
        }
      }
    }
  },
```

**Comment by Ben0189** [comment]
As per Tina team daily meeting on 26/8/2024 ,  we are reopening this issue due to the reappearance of a vulnerability. This might be a result of the numerous version upgrades and recent fixes.

**Comment by Ben0189** [comment]
![Image](https://github.com/user-attachments/assets/faddd33f-e3a6-4516-993b-07e1821a726e)

**Figure: Current vulnerability packages**

**Comment by Ben0189** [comment]
**Progress update**

vulnerable down from 5 -> 3

Leftover vulnerable packages 

- lodash.set - used in graphql packages, need further investigation for current set method usage in the package (we might need to replace lodash set)
- micromatch - not worth fixing it because it is under example project (which going to be clean up in future and will not be use by user)
- ws - not worth fixing it because it is under example project (which going to be clean up in future and will not be use by user)

**Comment by Ben0189** [comment]
Vulnerable packages down to 1 

Last packages lodash.set

PR for review 
https://github.com/tinacms/tinacms/pull/4754

**Comment by ncn-ssw** [comment]
✅ Done - PBI created for remaining package
